components/forms/_checkboxes/checkbox.scss

```diff
@@ -0,0 +1,104 @@
+.msds-checkbox {
+  margin: 20px;
+
+  &__input {
+    position: absolute;
+    opacity: 0;
+    + label {
+      position: relative;
+      cursor: pointer;
+      padding: 0;
+      display: flex;
+      align-items: flex-start;
+      -webkit-tap-highlight-color: transparent; //remove blue highlight on ios devices
+
+      &:before {
+        content: '';
+        margin-right: 20px;
+        display: inline-block;
+        vertical-align: text-top;
+        min-width: 24px;
+        width: 24px;
+        height: 24px;
+        background: $white;
+        transition: all 0.2s ease;
+        border-radius: 2px;
+        border: 1px solid $gray-5;
+      }
+    }
+
+    &:hover {
+      + label:before {
+        background: $gray-2;
+      }
+    }
+
+    &:focus {
+      + label:before {
+        box-shadow: 0 0 0 4px $rgba-3;
+        border: 1px solid $clear-blue;
+      }
+    }
+
+    &:checked {
+      + label:before {
+        border: 1px solid $clear-blue;
+        background: $clear-blue;
+      }
+    }
+
+    &:disabled,
+    &--disabled {
+      + label {
+        color: $gray-7;
+        cursor: auto;
+        &:before {
+          box-shadow: none;
+          background: $gray-3;
+          border: 1px solid $gray-5;
+        }
+      }
+    }
+
+    &--readonly,
+    &--readonly:checked {
+      pointer-events: none;
+
+      + label {
```

Comment:
I would add a HTML class to the label to avoid styling html elements just to be safe

Reply:
this is a modifier of the input filed which already has a class, however, I'll apply a specific class for read-only

Reply:
Thanks :) I think my concern was regarding line 67

Reply:
Oh didn't see that :) my bad. Label has been adjusted though :)